### PR TITLE
Issue/4/galaxycluster outline

### PR DESCRIPTION
**Pull request by m-aguena**
Working version of the GalaxyCluster object with internal management functions.

### CODE REVIEW

**Comment by cristobal-sifon** [comment]
also please add future imports re #29. It is much easier to do this when files are created than to keep track of incompatible files after the fact

File: clmm/tests/test_galaxycluster.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+test_dict = {'test%d'%i:True for i in range(3)}
+test_dict_diff = {'test%d'%i:False for i in range(3)}
+test_dict_sub = {'test%d'%i:True for i in range(2)}
+test_data = GCData(test_creator, test_dict,[1, 2, 3])
+
+from numpy import testing as tst
+
+def test_check_subdict():
+
+    tst.assert_equal(True, check_subdict(test_dict_sub, test_dict))
```
Comment (matthewkirby) [inline clmm/tests/test_galaxycluster.py]: `assert check_subdict(test_dict_sub, test_dict)`
`assert not check_subdict(test_dict, test_dict_sub)`

File: clmm/tests/test_galaxycluster.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    gc = GalaxyCluster(test_data)
+
+    tst.assert_equal([test_data], gc.find_data(test_creator, test_dict))
+    tst.assert_equal([test_data], gc.find_data(test_creator, test_dict_sub))
+    tst.assert_raises(ValueError, gc.find_data, test_creator, test_dict_diff)
+
+    tst.assert_equal(test_data, gc.find_data(test_creator, test_dict, exact=True))
+    tst.assert_raises(ValueError, gc.find_data, test_creator, test_dict_sub, exact=True)
+    tst.assert_raises(ValueError, gc.find_data, test_creator, test_dict_diff, exact=True)
+    pass
```
Comment (matthewkirby) [inline clmm/tests/test_galaxycluster.py]: take this out everywhere

File: clmm/tests/test_galaxycluster.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    tst.assert_raises(TypeError, gc.add_data, '')
+    tst.assert_raises(TypeError, gc.add_data, '', force=True)
+    tst.assert_equal(None, gc.add_data(test_data, force=True))
+
+    gc = GalaxyCluster()
+    tst.assert_equal(None, gc.add_data(test_data))
+    tst.assert_raises(ValueError, gc.add_data, test_data)
+    tst.assert_equal(None, gc.add_data(test_data, force=True))
+
+    #(self, incoming_data, force=False):
```
Comment (matthewkirby) [inline clmm/tests/test_galaxycluster.py]: take this out as well

File: clmm/galaxycluster.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                return
+        raise ValueError('incoming data not found in GalaxyCluster')
+
+    def _setup_dir(self, homelocal):
+        """
+        Checks if directory exists, otherwise makes it
+        """
+        pass
+
+    def read_GC(self, filename, lookup_creators=None, lookup_specs=None):
```
Comment (matthewkirby) [inline clmm/galaxycluster.py]: Make this raise an error

File: clmm/galaxycluster.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        Reads in a pickled GalaxyCluster's data from saved versions
+        """
+        pass
+
+    def write_GC(self, filename, lookup_creator=None, lookup_specs=None):
```
Comment (matthewkirby) [inline clmm/galaxycluster.py]: Make this raise an error

**Comment by matthewkirby** [comment]
All requested changes changed!

**Pull request merged**